File: doc/codes/csharp/examples/api/compute/pattern/twin-trap.cs
using System.Numerics;
using AUTD3;
using static AUTD3.Units;

namespace DocSamples.ApiComputePatternTwinTrap;

internal static class Sample
{
    internal static void Run()
    {
        var geometry = new Geometry(new[] { new Autd3(Vector3.Zero) });

        var target = geometry.Center + new Vector3(0.0f, 0.0f, 150.0f);
        var normal = Vector3.UnitX;
        var wavelength = Pattern.Wavelength(340.0f * m / s);
        var dst = geometry.PatternBuffer();

        // ANCHOR: api
        Pattern.TwinTrap(geometry, target, normal, wavelength, dst);
        // ANCHOR_END: api
    }
}

File: doc/codes/rust/examples/api/compute/pattern/twin-trap.rs
use autd3_rs::geometry::{Autd3, Geometry, Vector3, offset};
use autd3_rs::units::{m, mm, s};
use autd3_rs_pattern::{twin_trap, wavelength};

fn main() {
    let geometry = Geometry::new(vec![Autd3::default()]);

    let target = geometry.center() + offset(0.0 * mm, 0.0 * mm, 150.0 * mm);
    let normal = Vector3::x_axis();
    let wavelength = wavelength(340.0 * m / s);
    let mut dst = geometry.pattern_buffer();

    // ANCHOR: api
    twin_trap(&geometry, target, normal, wavelength, &mut dst);
    // ANCHOR_END: api
}
